release/models/platform/openconfig-platform.yang

```diff
@@ -663,6 +669,80 @@ module openconfig-platform {
     }
   }
 
+  grouping platform-health-top {
+    description
+      "Grouping containing health-related parameters.";
+
+    container healthz {
+      description
+        "The health of the component. The paramaters within this
+        container indicate the status of the component beyond whether
+        it is operationally up or down. When a signal is received
+        that a component is in an unhealthy state the gNOI.Healthz
+        service can be used to retrieve further diagnostic information
+        relating to the component.
+
+        The contents of this directory relate only to the specific
+        component that it is associated with. In the case that child
+        components become unhealthy and this causes a parent component
+        to be unhealthy, the new unhealthy status should be reported at
+        both components, such that an interested system can take the
+        relevant actions (e.g., retrieve the Healthz output, or
+        apply mitigation actions).";
+      reference
+        "https://https://github.com/openconfig/gnoi/tree/master/healthz";
+
+      container state {
+        config false;
+        description
+          "Operational state parameters relating to component health.";
+        uses platform-health-state;
+      }
+    }
+  }
+
+  grouping platform-health-state {
+    description
+      "Operational state parmaeters relating to a platform component's
+      health.";
+
+    leaf status {
+      type enumeration {
+        enum HEALTHY {
+          description
+            "The component is in a HEALTHY state, and is operating
+            within the expected parameters.";
+        }
+
+        enum FAILED {
```

Comment:
enum Status {
  STATUS_UNSPECIFIED = 0;   // The path doesn't support healthz
  STATUS_HEALTHY = 1;       // The path is healthy
  STATUS_UNHEALTHY = 2;     // The path is unhealthy
}

why not unhealthly vs failed
also i think you do need to model unspecified?

Reply:
We discussed and mentioned `UNSPECIFIED` will mean explicitly that the device has not checked the status of the component.